src/setup-k8s/templates/fabric-k8s/scripts/base-functions.sh

```diff
@@ -36,79 +39,83 @@ deployPeer() {
 
-  printItalics "Joining 'my-channel1' on  Org1/peer1" "U1F638"
+        kubectl hlf ordnode join --block=$CONFIG_DIR/<%= channel.name %>.block --name=<%= channel.ordererHead.orgName.toLowerCase() %>-node --namespace=$NAMESPACE --identity=$CONFIG_DIR/admin-tls-ordservice.yaml
 
-  retry 3 kubectl hlf channel join --name=$CHANNEL_NAME --config=$CONFIG_DIR/org1.yaml --user=$ADMIN_USER -p=$ORG-peer1.$NAMESPACE
+        kubectl hlf ca register --name=<%= org.name.toLowerCase() %>-<%= org.ca.prefix %> --user=$<%= org.ca.caAdminNameVar %> --secret=$<%= org.ca.caAdminPassVar %> --type=admin --enroll-id <%= org.name.toLowerCase() %> --enroll-secret=$<%= org.ca.caAdminPassVar %> --mspid <%= org.mspName %> &&
+        kubectl hlf ca enroll --name=<%= org.name.toLowerCase() %>-<%= org.ca.prefix %> --user=$<%= org.ca.caAdminNameVar %> --secret=$<%= org.ca.caAdminPassVar %> --mspid <%= org.mspName %> --ca-name ca --output $CONFIG_DIR/peer-<%= org.name %>.yaml &&
+        kubectl hlf inspect --output $CONFIG_DIR/<%= org.name %>.yaml -o <%= org.mspName %> -o <%= channel.ordererHead.orgMspName %> &&
+        kubectl hlf utils adduser --userPath=$CONFIG_DIR/peer-<%= org.name %>.yaml --config=$CONFIG_DIR/<%= org.name %>.yaml --username=$<%= org.ca.caAdminNameVar %> --mspid=<%= org.mspName %> &&
 
-  # add anchor peer
+      <% org.peers.forEach((peer) => { %>
+        printItalics "Joining '<%= channel.name %>' on  <%= org.name.toLowerCase() %>/<%= peer.name %>" "U1F638"
+        retry 3 kubectl hlf channel join --name=<%= channel.name %> --config=$CONFIG_DIR/<%= org.name %>.yaml --user=$<%= org.ca.caAdminNameVar %> -p=<%= peer.name %>.$NAMESPACE
+      <% }) %>
 
-  printItalics "Electing on Org1/peer0 as Anchor peer" "U1F638"
+      # add anchor peers
+      <% org.anchorPeers.forEach((anchorpeer) => { %>
+        printItalics "Electing on <%= org.name.toLowerCase() %>/<%= anchorpeer.name %> as Anchor peer" "U1F638"
+        kubectl hlf channel addanchorpeer --channel=<%= channel.name %> --config=$CONFIG_DIR/<%= org.name %>.yaml --user=$<%= org.ca.caAdminNameVar %> --peer=<%= anchorpeer.name %>.$NAMESPACE
+      <% }) %>
 
-  kubectl hlf channel addanchorpeer --channel=$CHANNEL_NAME --config=$CONFIG_DIR/org1.yaml --user=$ADMIN_USER --peer=$ORG-peer0.$NAMESPACE
+      <% }) -%>
+    <% }) -%>
 }
 
-installChaincodes() {
-  printItalics "Building chaincode $CHAINCODE_NAME" "U1F618"
-
-  buildAndInstallChaincode "$CHAINCODE_NAME" "0" "$CHAINCODE_LANG" "$CHAINCODES_BASE_DIR/./chaincodes/chaincode-kv-node" "$CHAINCODE_VERSION"
-  buildAndInstallChaincode "$CHAINCODE_NAME" "1" "$CHAINCODE_LANG" "$CHAINCODES_BASE_DIR/./chaincodes/chaincode-kv-node" "$CHAINCODE_VERSION"
 
-  approveChaincode "$CHAINCODE_NAME" "1" "$CHAINCODE_VERSION" "$CHANNEL_NAME"
-
-  printItalics "Committing chaincode '$CHAINCODE_NAME' on channel '$CHANNEL_NAME' as '$ORG'" "U1F618"
-  commitChaincode "$CHAINCODE_NAME" "1" "$CHAINCODE_VERSION" "$CHANNEL_NAME"
+installChaincodes() {
```

Comment:
We might improve this one - if we don't have any chaincodes then methods body will be empty and the startup will fail.

Reply:
I think we could improve this by removing the chaincode setup from startup: ``./fablo.sh up`` and only install chaincodes with ``./fablo.sh chaincodes install``